File: Tutorials/Tuckshop/Client/Tuckshop.WebApp/src/Domain/Models/Wallets/Commands/WithdrawFromWallet.ts
import { Attributes, ModelBase, Rules, Validation } from '@singularsystems/neo-core';
export default class WithdrawFromWallet extends ModelBase {
    static typeName = "WithdrawFromWallet";

    constructor() {
        super();
        this.makeObservable();
    }

    @Rules.Required()
    public customerId: number = 0;

    @Attributes.Float()
    public amount: number = 0;

    // Client only properties / methods

    protected static addBusinessRules(rules: Validation.Rules<WithdrawFromWallet>) {
        super.addBusinessRules(rules);
    }

    public toString(): string {
        if (this.isNew) {
            return "New withdraw from wallet";
        } else {
            return "Withdraw From Wallet";
        }
    }
}

File: Tutorials/Tuckshop/Server/Tuckshop.Domain/Tuckshop.Core.Models/Models/Wallets/Commands/WithdrawFromWallet.cs
namespace Tuckshop.Core.Models.Wallets.Commands
{
  using Neo.Model;
  using Neo.Model.Validation;
  using System.ComponentModel.DataAnnotations;
  using System.ComponentModel.DataAnnotations.Schema;

  /// <summary>
  /// Class Representing a withdrawal from a customer's wallet.
  /// </summary>
  public class WithdrawFromWallet : ModelBase<WithdrawFromWallet>
  {
    [Required]
    public int CustomerId { get; set; }

    /// <summary>
    /// Gets and sets the Amount to withdraw.
    /// </summary>
    [Column(TypeName = "money")]
    public decimal Amount { get; set; }

    /// <inheritdoc/>
    protected override void AddBusinessRules(ValidationRules<WithdrawFromWallet> rules)
    {
      rules.FailWhen(c => c.Amount > 500m, "Withdrawal amount cannot exceed R500 per transaction.");
    }
  }
}
